server/src/main/java/org/opensearch/cluster/health/ClusterStateHealth.java

```diff
@@ -229,6 +237,7 @@ public void writeTo(final StreamOutput out) throws IOException {
         out.writeVInt(unassignedShards);
         out.writeVInt(numberOfNodes);
         out.writeVInt(numberOfDataNodes);
+        out.writeBoolean(hasDiscoveredMaster);
```

Comment:
Similar version check needed here on the output stream as well.

Reply:
Done.